gnat: 9 -> 11; update AdaCore packages

sternenseemann opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

* Closes #143714.

* GNAT 11 is now required to build AdaCore's packages and we can also compile everything in nixpkgs so far using it, so we may as well upgrade to the latest version.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

sternenseemann committed:
gnat: 9 -> 11

Update the default GNAT version from 9 to 11, as GNAT >= 11 is required
to compile the 22.* AdaCore libraries.

To allow this, we need to pick a patch from ghdl's master fixing a
compilation problem with GNAT 11.

sternenseemann committed:
xmlada: 21.0.0 -> 22.0.0

sternenseemann committed:
gprbuild, gprbuild-boot: 21.0.0 -> 22.0.0

sternenseemann committed:
gnatcoll-*: 21.0.0 -> 22.0.0

SuperSandro2000 commented on pkgs/development/libraries/ada/gnatcoll/bindings.nix:
```diff
@@ -68,7 +68,7 @@ stdenv.mkDerivation rec {
   buildPhase = ''
     runHook preBuild
-    python3 ${component}/setup.py build $buildFlags
+    python3 ${component}/setup.py build --prefix $out $buildFlags
```
```suggestion
    ${python.interpeter} ${component}/setup.py build --prefix $out $buildFlags
```

sternenseemann replied:
Done.

sternenseemann committed:
gnatcoll-*: get python binary name from derivation

vcunat commented:
I don't know Ada stuff, but this sounds OK for 21.11.

sternenseemann committed:
emacs.pkgs.ada-mode: use latest wisi

By upgrading the AdaCore libraries to their 22.*.* versions and GNAT to
version 11 we can finally upgrade wisi beyond 3.1.5 and probably
(hopefully) stay on latest for some time.

sternenseemann commented:
I think I'm gonna merge this early next week, if there are no concerns, given that Ada doesn't have a lot upstream (that is nixpkgs itself) stakeholders, I think I'd be struggling to find someone else to pull the trigger :p

Downstream concerns should be covered by release notes entry in principle.
